Makefile

```diff
@@ -175,6 +175,48 @@ docs:
 	@echo "* $(CLR_BLUE)$(SRC_TAGGED_URL_BASE)/docs/user-guide.md$(CLR_END)"
 	@echo ""
 
+verify_tooling: .venv
+	@( \
+		. .venv/bin/activate; \
+		python3 --version &> /dev/null && \
+		( \
+			python3 -c "import sys; sys.version_info < (3,11) and sys.exit(1)" || \
```

Comment:
I'm not a fan of where the minimum version is stored and compared (e.g. in the middle of the make file). Not sure what a better approach would be, but this may be hard to find for another maintainer.

Reply:
Good point, I will try to move it to the top somehow.
I'm trying to keep the Makefile one single file so it is easy to use with older versions of ADF as well.

Reply:
I've updated the Makefile to include the version number definition at the top as requested.